From 5459bed660e5b97eb2d549e1c4ab22a4de46e542 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Mon, 20 Jun 2016 13:16:23 +0300
Subject: [PATCH 0573/2241] fix: sar: ap806: fix kernel crash when CPU
 frequency is not supported

Remove the return from clk_init function when sample at reset CPU clock
frequency is not supported.
The issue found when loading Kernel with unsupported CPU frequency,
the function return without configure the MSS clock, and UART baud-rate
not configured correctly.
Setting CPU clock to 0 (zoro) is invalid, and might lead to unexpected
unit behavior, but avoiding kernel crash

Change-Id: I80ba550a819f3ad7c69626820db58642b6bc6426
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30566
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Neta Zur Hershkovits <neta@marvell.com>
---
 drivers/clk/mvebu/ap806-system-controller.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/mvebu/ap806-system-controller.c b/drivers/clk/mvebu/ap806-system-controller.c
index ccca3af..3235b37 100644
--- a/drivers/clk/mvebu/ap806-system-controller.c
+++ b/drivers/clk/mvebu/ap806-system-controller.c
@@ -69,8 +69,12 @@ static void __init ap806_syscon_clk_init(struct device_node *np)
 		cpuclk_freq = 1300;
 		break;
 	default:
+		/* set cpuclk_freq as invalid value to continue and
+		** configure the MSS clock (used to calculate the
+		** baudrate of the UART
+		*/
+		cpuclk_freq = 0;
 		pr_err("invalid SAR value\n");
-		return;
 	}
 
 	/* Convert to hertz */
-- 
2.7.4

